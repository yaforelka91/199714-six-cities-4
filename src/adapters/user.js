const adaptUser = (user) => {
  return {
    id: user.id,
    email: user.email,
    name: user.name,
    picture: user.avatar_url,
    isSuper: user.is_pro,
  };
};

export default adaptUser;
