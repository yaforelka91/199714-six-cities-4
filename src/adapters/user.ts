import User from '../interfaces/user';

const adaptUser = (user): User => ({
  id: user.id,
  email: user.email,
  name: user.name,
  picture: user.avatar_url,
  isSuper: user.is_pro,
});

export default adaptUser;
