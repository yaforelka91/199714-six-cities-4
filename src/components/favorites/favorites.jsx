import React, {Fragment} from 'react';
import {connect} from 'react-redux';
import {favoritesTypes} from '../../types/types.js';
import CityListItem from '../city-list-item/city-list-item.jsx';
import OfferList from '../offer-list/offer-list.jsx';
import {CardView} from '../../const.js';
import {ActionCreator} from '../../reducer/catalog/catalog.js';

const Favorites = ({offers, onCityNameClick}) => {
  return (
    <main className={`page__main page__main--favorites${offers.length === 0 ? ` page__main--favorites-empty` : ``}`}>
      <div className="page__favorites-container container">
        <section className={`favorites${offers.length === 0 ? ` favorites--empty` : ``}`}>
          {offers.length === 0 ?
            <Fragment>
              <h1 className="visually-hidden">Favorites (empty)</h1>
              <div className="favorites__status-wrapper">
                <b className="favorites__status">Nothing yet saved.</b>
                <p className="favorites__status-description">Save properties to narrow down search or plan yor future trips.</p>
              </div>
            </Fragment> :
            <Fragment>
              <h1 className="favorites__title">Saved listing</h1>
              <ul className='favorites__list'>
                {
                  offers.map((localOffer) => {
                    const currentCity = Object.keys(localOffer)[0];

                    return (
                      <li className='favorites__locations-items' key={currentCity}>
                        <div className="favorites__locations locations locations--current">
                          <CityListItem
                            tagName='div'
                            city={currentCity}
                            onCityNameClick={() => {
                              onCityNameClick(currentCity);
                            }}
                          />
                        </div>
                        <OfferList
                          offers={localOffer[currentCity]}
                          className='favorites__places'
                          viewMode={CardView.FAVORITES}
                        />
                      </li>
                    );
                  })
                }
              </ul>
            </Fragment>
          }
        </section>
      </div>
    </main>
  );
};

Favorites.propTypes = favoritesTypes;

const mapDispatchToProps = (dispatch) => ({
  onCityNameClick(city) {
    dispatch(ActionCreator.changeCity(city));
  },
});

export {Favorites};
export default connect(null, mapDispatchToProps)(Favorites);
