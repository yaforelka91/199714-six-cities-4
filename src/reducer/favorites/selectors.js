import NameSpace from '../name-space.js';
import {createSelector} from 'reselect';
import {getOffers} from '../data/selectors.js';

const NAME_SPACE = NameSpace.FAVORITES;

export const getFavorites = (state) => {
  return state[NAME_SPACE].favoriteList;
};

export const getFavoriteOffers = createSelector(
    getOffers,
    getFavorites,
    (allOffers, favoriteOffers) => {
      const reducedFavoriteOffers = favoriteOffers.map((offer) => offer.isFavorite);

      return allOffers.filter((offer) => {
        return reducedFavoriteOffers.includes(offer.isFavorite);
      });
    }
);

export const getGroupedCities = createSelector(
    getFavoriteOffers,
    (offers) => {
      return [...new Set(offers
  .map((offer) => offer.city.name)
  .map((cityObject) => JSON.stringify(cityObject))
      )]
  .map((cityString) => JSON.parse(cityString));
    }
);

export const getGroupedFavoriteOffers = createSelector(
    getGroupedCities,
    getFavoriteOffers,
    (cities, offers) => {
      return cities.map((city) => {
        return {
          [city]: offers.filter((offer) => {
            return offer.city.name === city;
          }),
        };
      });
    }
);
