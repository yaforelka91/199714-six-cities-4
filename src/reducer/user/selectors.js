import NameSpace from '../name-space.js';

const NAME_SPACE = NameSpace.USER;

export const getAuthorizationStatus = (state) => {
  return state[NAME_SPACE].authorizationStatus;
};

export const getAuthorizationProgress = (state) => {
  return state[NAME_SPACE].isAuthorizationInProgress;
};

export const getUserData = (state) => {
  return state[NAME_SPACE].userData;
};

export const getError = (state) => {
  return state[NAME_SPACE].errorType;
};
