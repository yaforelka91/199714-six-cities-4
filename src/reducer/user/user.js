import {extend} from '../../utils.js';
import adaptUser from '../../adapters/user.js';
import adaptError from '../../adapters/error.js';
import {Operation as DataOperation} from '../data/data.js';
import {Error} from '../../api.js';

const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

const initialState = {
  authorizationStatus: AuthorizationStatus.NO_AUTH,
  userData: {},
  isAuthorizationInProgress: true,
  errorType: ``,
};

const ActionType = {
  REQUIRED_AUTHORIZATION: `REQUIRED_AUTHORIZATION`,
  SET_USER_DATA: `SET_USER_DATA`,
  CHANGE_PROGRESS_STATUS: `CHANGE_PROGRESS_STATUS`,
  CATCH_ERROR: `CATCH_ERROR`,
};

const ActionCreator = {
  requireAuthorization: (status) => {
    return {
      type: ActionType.REQUIRED_AUTHORIZATION,
      payload: status,
    };
  },
  setUserData: (authData) => {
    return {
      type: ActionType.SET_USER_DATA,
      payload: authData,
    };
  },
  changeProgressStatus: (isLoaded) => {
    return {
      type: ActionType.CHANGE_PROGRESS_STATUS,
      payload: isLoaded,
    };
  },
  catchError: (errorMessage) => {
    return {
      type: ActionType.CATCH_ERROR,
      payload: errorMessage,
    };
  },
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.REQUIRED_AUTHORIZATION:
      return extend(state, {
        authorizationStatus: action.payload,
      });

    case ActionType.SET_USER_DATA:
      return extend(state, {
        userData: action.payload,
      });

    case ActionType.CHANGE_PROGRESS_STATUS:
      return extend(state, {
        isAuthorizationInProgress: action.payload,
      });

    case ActionType.CATCH_ERROR:
      return extend(state, {
        errorType: action.payload,
      });
  }

  return state;
};

const Operation = {
  checkAuth: () => (dispatch, getState, api) => {
    return api.get(`/login`)
      .then((response) => {
        dispatch(ActionCreator.changeProgressStatus(false));
        dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH));
        dispatch(ActionCreator.setUserData(adaptUser(response.data)));
      })
      .catch((err) => {
        dispatch(ActionCreator.changeProgressStatus(false));
        throw err;
      });
  },
  login: (authData) => (dispatch, getState, api) => {
    return api.post(`/login`, {
      email: authData.login,
      password: authData.password,
    })
      .then((response) => {
        dispatch(ActionCreator.changeProgressStatus(false));
        dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH));
        dispatch(ActionCreator.setUserData(adaptUser(response.data)));
        dispatch(ActionCreator.catchError(``));
        dispatch(DataOperation.loadOffers());
      })
      .catch((err) => {
        dispatch(ActionCreator.changeProgressStatus(false));

        if (err.response.status === Error.BAD_REQUEST) {
          dispatch(ActionCreator.catchError(adaptError(err.response.data.error)));
          return;
        }

        throw err;
      });
  },
};

export {
  ActionCreator,
  ActionType,
  AuthorizationStatus,
  Operation,
  reducer,
};
