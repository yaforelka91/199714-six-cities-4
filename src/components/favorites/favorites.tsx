import React, {PureComponent} from 'react';
import {connect} from 'react-redux';
import CityListItem from '../city-list-item/city-list-item';
import OfferList from '../offer-list/offer-list';
import {CardView} from '../../types';
import {ActionCreator} from '../../reducer/catalog/catalog';
import {Operation} from '../../reducer/favorites/favorites';
import {getGroupedFavoriteOffers} from '../../reducer/favorites/selectors';
import Offer from '../../interfaces/offer';

type Props= {
  offers: {
    [key: string]: Offer[];
  }[];
  onCityNameClick: (city: string) => void;
  onFavoritesRequest: () => void;
};

class Favorites extends PureComponent<Props, {}> {
  props: Props;

  componentDidMount() {
    this.props.onFavoritesRequest();
  }

  render() {
    const {offers, onCityNameClick} = this.props;

    return (
      <main className={`page__main page__main--favorites${offers.length === 0 ? ` page__main--favorites-empty` : ``}`}>
        <div className="page__favorites-container container">
          <section className={`favorites${offers.length === 0 ? ` favorites--empty` : ``}`}>
            {offers.length === 0 ?
              <>
                <h1 className="visually-hidden">Favorites (empty)</h1>
                <div className="favorites__status-wrapper">
                  <b className="favorites__status">Nothing yet saved.</b>
                  <p className="favorites__status-description">Save properties to narrow down search or plan yor future trips.</p>
                </div>
              </> :
              <>
                <h1 className="favorites__title">Saved listing</h1>
                <ul className='favorites__list'>
                  {
                    offers.map((localOffer) => {
                      const currentCity = Object.keys(localOffer)[0];

                      return (
                        <li className='favorites__locations-items' key={currentCity}>
                          <div className="favorites__locations locations locations--current">
                            <CityListItem
                              tagName='div'
                              city={currentCity}
                              onCityNameClick={() => onCityNameClick(currentCity)}
                            />
                          </div>
                          <OfferList
                            offers={localOffer[currentCity]}
                            className='favorites__places'
                            viewMode={CardView.FAVORITES}
                          />
                        </li>
                      );
                    })
                  }
                </ul>
              </>
            }
          </section>
        </div>
      </main>
    );
  }
}

const mapStateToProps = (state) => ({
  offers: getGroupedFavoriteOffers(state),
});

const mapDispatchToProps = (dispatch) => ({
  onCityNameClick(city) {
    dispatch(ActionCreator.changeCity(city));
  },
  onFavoritesRequest() {
    dispatch(Operation.loadFavorites());
  }
});

export {Favorites};
export default connect(mapStateToProps, mapDispatchToProps)(Favorites);
