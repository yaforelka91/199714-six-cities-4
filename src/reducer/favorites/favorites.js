import {extend} from '../../utils';
import adaptOffer from '../../adapters/offers.js';
import {ActionCreator as DataActionCreator} from '../data/data.js';
import {getOffers} from '../data/selectors.js';

const initialState = {
  favorites: {},
  favoriteList: [],
};

const ActionType = {
  TOGGLE_FAVORITE: `TOGGLE_FAVORITE`,
  GET_FAVORITE: `GET_FAVORITE`,
};

const ActionCreator = {
  toggleFavorite: (hotel) => ({
    type: ActionType.TOGGLE_FAVORITE,
    payload: hotel,
  }),
  getFavorite: (offers) => ({
    type: ActionType.GET_FAVORITE,
    payload: offers,
  }),
};

const Operation = {
  changeFavoriteStatus: (hotel) => (dispatch, getState, api) => {
    return api.post(`/favorite/${hotel.id}/${hotel.isFavorite}`, {
      hotel
    })
      .then((response) => {
        const adaptedOffer = adaptOffer(response.data);
        dispatch(ActionCreator.toggleFavorite(adaptedOffer));

        const index = getOffers(getState()).map((offer) => offer.id).indexOf(adaptedOffer.id);
        getOffers(getState())[index] = adaptedOffer;
        dispatch(DataActionCreator.loadOffers(getOffers(getState()).slice()));
      })
    .catch((err) => {
      throw err;
    });
  },
  loadFavorites: () => (dispatch, getState, api) => {
    return api.get(`/favorite`)
      .then((response) => {
        const adaptedOffers = response.data.map((offer) => {
          return adaptOffer(offer);
        });

        dispatch(ActionCreator.getFavorite(adaptedOffers));
      })
    .catch((err) => {
      throw err;
    });
  }
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.TOGGLE_FAVORITE:
      return extend(state, {
        favorites: action.payload,
      });

    case ActionType.GET_FAVORITE:
      return extend(state, {
        favoriteList: action.payload,
      });
  }
  return state;
};

export {
  reducer,
  ActionCreator,
  ActionType,
  Operation,
};
