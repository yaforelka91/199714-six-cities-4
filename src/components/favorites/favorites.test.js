import React from 'react';
import renderer from 'react-test-renderer';
import {Favorites} from './favorites.jsx';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import NameSpace from '../../reducer/name-space.js';
import {Router} from 'react-router-dom';
import history from '../../history.js';
import {AuthorizationStatus} from '../../reducer/user/user.js';

const mockStore = configureStore([]);

const mock = {
  offersList: [
    {
      'city 1': [
        {
          city: {
            name: `city 1`,
            coords: [0, 0],
            zoom: 1,
          },
          id: 1,
          coords: [1, 2],
          offerZoom: 12,
          title: `Beautiful & luxurious apartment at great location`,
          description: [
            `A quiet cozy and picturesque that hides behind a 
    river by the unique lightness of Amsterdam.
    The building is green and from 18th century.`,
            `An independent House, strategically located 
    between Rembrand Square and National Opera, 
    but where the bustle of the city comes to rest 
    in this alley flowery and colorful.`
          ],
          picture: `http://placeimg.com/260/200/arch`,
          pictures: [
            `http://placeimg.com/260/200/arch`,
            `http://placeimg.com/260/200/arch`,
            `http://placeimg.com/260/200/arch`,
            `http://placeimg.com/260/200/arch`,
            `http://placeimg.com/260/200/arch`,
            `http://placeimg.com/260/200/arch`
          ],
          price: 120,
          type: `Apartment`,
          isPremium: true,
          isFavorite: false,
          rating: 4.1,
          bedrooms: 3,
          guests: 4,
          services: [
            `Wi-Fi`,
            `Washing machine`,
            `Towels`,
            `Heating`,
            `Coffee machine`,
            `Baby seat`,
            `Kitchen`,
            `Dishwasher`,
            `Cabel TV`,
            `Fridge`,
          ],
          host: {
            id: 3,
            name: `Angelina`,
            picture: `http://placekitten.com/74/74`,
            isSuper: true,
          },
        }
      ]
    },
    {
      'city 2': [
        {
          city: {
            name: `city 2`,
            coords: [0, 0],
            zoom: 1,
          },
          id: 2,
          coords: [3, 4],
          offerZoom: 12,
          title: `Beautiful & luxurious apartment at great location`,
          description: [
            `A quiet cozy and picturesque that hides behind a 
        river by the unique lightness of Amsterdam.
        The building is green and from 18th century.`,
            `An independent House, strategically located 
        between Rembrand Square and National Opera, 
        but where the bustle of the city comes to rest 
        in this alley flowery and colorful.`
          ],
          picture: `http://placeimg.com/260/200/arch`,
          pictures: [
            `http://placeimg.com/260/200/arch`,
            `http://placeimg.com/260/200/arch`,
            `http://placeimg.com/260/200/arch`,
            `http://placeimg.com/260/200/arch`,
            `http://placeimg.com/260/200/arch`,
            `http://placeimg.com/260/200/arch`
          ],
          price: 120,
          type: `Apartment`,
          isPremium: true,
          isFavorite: false,
          rating: 4.1,
          bedrooms: 3,
          guests: 4,
          services: [
            `Wi-Fi`,
            `Washing machine`,
            `Towels`,
            `Heating`,
            `Coffee machine`,
            `Baby seat`,
            `Kitchen`,
            `Dishwasher`,
            `Cabel TV`,
            `Fridge`,
          ],
          host: {
            id: 5,
            name: `Angelina`,
            picture: `http://placekitten.com/74/74`,
            isSuper: true,
          },
        },
      ]
    }
  ],
};

describe(`FavoritesSnapshot`, () => {
  const {offersList} = mock;

  it(`should render Favorites`, () => {
    const store = mockStore({
      [NameSpace.USER]: {
        authorizationStatus: AuthorizationStatus.AUTH,
      },
    });

    const tree = renderer.create(
        <Provider store={store}>
          <Router history={history}>
            <Favorites
              offers={offersList}
              onFavoritesRequest={() => {}}
              onCityNameClick={() => {}}
            />
          </Router>
        </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`should render empty Favorites`, () => {
    const store = mockStore({
      [NameSpace.USER]: {
        authorizationStatus: AuthorizationStatus.AUTH,
      },
    });

    const tree = renderer.create(
        <Provider store={store}>
          <Router history={history}>
            <Favorites
              offers={[]}
              onFavoritesRequest={() => {}}
              onCityNameClick={() => {}}
            />
          </Router>
        </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
