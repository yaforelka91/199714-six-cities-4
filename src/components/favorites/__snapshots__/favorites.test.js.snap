// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesSnapshot should render Favorites 1`] = `
<main
  className="page__main page__main--favorites"
>
  <div
    className="page__favorites-container container"
  >
    <section
      className="favorites"
    >
      <h1
        className="favorites__title"
      >
        Saved listing
      </h1>
      <ul
        className="favorites__list"
      >
        <li
          className="favorites__locations-items"
        >
          <div
            className="favorites__locations locations locations--current"
          >
            <div
              className="locations__item"
            >
              <a
                className="locations__item-link"
                href="/"
                onClick={[Function]}
              >
                <span>
                  city 1
                </span>
              </a>
            </div>
          </div>
          <div
            className="favorites__places"
          >
            <article
              className="favorites__card place-card"
              onMouseEnter={[Function]}
            >
              <div
                className="place-card__mark"
              >
                <span>
                  Premium
                </span>
              </div>
              <div
                className="favorites__image-wrapper place-card__image-wrapper"
              >
                <a
                  href="/offer/1"
                  onClick={[Function]}
                >
                  <img
                    alt="Place image"
                    className="place-card__image"
                    height="200"
                    src="http://placeimg.com/260/200/arch"
                    width="260"
                  />
                </a>
              </div>
              <div
                className="favorites__card-info place-card__info"
              >
                <div
                  className="place-card__price-wrapper"
                >
                  <div
                    className="place-card__price"
                  >
                    <b
                      className="place-card__price-value"
                    >
                      €
                      120
                    </b>
                    <span
                      className="place-card__price-text"
                    >
                      / night
                    </span>
                  </div>
                  <button
                    className="place-card__bookmark-button button"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    <svg
                      className="place-card__bookmark-icon"
                      height="19"
                      width="18"
                    >
                      <use
                        xlinkHref="#icon-bookmark"
                      />
                    </svg>
                    <span
                      className="visually-hidden"
                    >
                      To bookmarks
                    </span>
                  </button>
                </div>
                <div
                  className="place-card__rating rating"
                >
                  <div
                    className="place-card__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "80%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  className="place-card__name"
                >
                  <a
                    href="/offer/1"
                    onClick={[Function]}
                  >
                    Beautiful & luxurious apartment at great location
                  </a>
                </h2>
                <p
                  className="place-card__type"
                >
                  Apartment
                </p>
              </div>
            </article>
          </div>
        </li>
        <li
          className="favorites__locations-items"
        >
          <div
            className="favorites__locations locations locations--current"
          >
            <div
              className="locations__item"
            >
              <a
                className="locations__item-link"
                href="/"
                onClick={[Function]}
              >
                <span>
                  city 2
                </span>
              </a>
            </div>
          </div>
          <div
            className="favorites__places"
          >
            <article
              className="favorites__card place-card"
              onMouseEnter={[Function]}
            >
              <div
                className="place-card__mark"
              >
                <span>
                  Premium
                </span>
              </div>
              <div
                className="favorites__image-wrapper place-card__image-wrapper"
              >
                <a
                  href="/offer/2"
                  onClick={[Function]}
                >
                  <img
                    alt="Place image"
                    className="place-card__image"
                    height="200"
                    src="http://placeimg.com/260/200/arch"
                    width="260"
                  />
                </a>
              </div>
              <div
                className="favorites__card-info place-card__info"
              >
                <div
                  className="place-card__price-wrapper"
                >
                  <div
                    className="place-card__price"
                  >
                    <b
                      className="place-card__price-value"
                    >
                      €
                      120
                    </b>
                    <span
                      className="place-card__price-text"
                    >
                      / night
                    </span>
                  </div>
                  <button
                    className="place-card__bookmark-button button"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    <svg
                      className="place-card__bookmark-icon"
                      height="19"
                      width="18"
                    >
                      <use
                        xlinkHref="#icon-bookmark"
                      />
                    </svg>
                    <span
                      className="visually-hidden"
                    >
                      To bookmarks
                    </span>
                  </button>
                </div>
                <div
                  className="place-card__rating rating"
                >
                  <div
                    className="place-card__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "80%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  className="place-card__name"
                >
                  <a
                    href="/offer/2"
                    onClick={[Function]}
                  >
                    Beautiful & luxurious apartment at great location
                  </a>
                </h2>
                <p
                  className="place-card__type"
                >
                  Apartment
                </p>
              </div>
            </article>
          </div>
        </li>
      </ul>
    </section>
  </div>
</main>
`;

exports[`FavoritesSnapshot should render empty Favorites 1`] = `
<main
  className="page__main page__main--favorites page__main--favorites-empty"
>
  <div
    className="page__favorites-container container"
  >
    <section
      className="favorites favorites--empty"
    >
      <h1
        className="visually-hidden"
      >
        Favorites (empty)
      </h1>
      <div
        className="favorites__status-wrapper"
      >
        <b
          className="favorites__status"
        >
          Nothing yet saved.
        </b>
        <p
          className="favorites__status-description"
        >
          Save properties to narrow down search or plan yor future trips.
        </p>
      </div>
    </section>
  </div>
</main>
`;
